### PR TITLE
Updates 20210204 MLKit new artifacts + GPS ML

### PR DESCRIPTION
**Pull request by moljac**
### Google Play Services Version (eg: 8.4.0):

GPS + MLKit manual updates + new artifacts

- com.google.android.gms:play-services-mlkit-barcode-scanning - 16.1.0 -> 116.1.0.1
  - [breaking.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950853/Xamarin.GooglePlayServices.MLKit.BarcodeScanning.dll.breaking.md.txt)
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950854/Xamarin.GooglePlayServices.MLKit.BarcodeScanning.dll.diff.md.txt)
- com.google.android.gms:play-services-mlkit-face-detection - 16.1.0 -> 116.1.0.1
  - [breaking.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950870/Xamarin.GooglePlayServices.MLKit.FaceDetection.dll.breaking.md.txt)
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950871/Xamarin.GooglePlayServices.MLKit.FaceDetection.dll.diff.md.txt)
- com.google.android.gms:play-services-mlkit-text-recognition - 16.0.0
  - [breaking.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950885/Xamarin.GooglePlayServices.MLKit.Text.Recognition.dll.breaking.md.txt)
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950886/Xamarin.GooglePlayServices.MLKit.Text.Recognition.dll.diff.md.txt)
- com.google.mlkit:common - 16.0.0 -> 16.1.0
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950896/Xamarin.Google.MLKit.Common.dll.diff.md.txt)
- com.google.mlkit:barcode-scanning - 16.0.0 -> 16.0.1
  - No public API changes
- com.google.mlkit:digital-ink-recognition - 16.0.0
  - [breaking.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950907/Xamarin.Google.MLKit.DigitalInk.Recognition.dll.breaking.md.txt)
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950908/Xamarin.Google.MLKit.DigitalInk.Recognition.dll.diff.md.txt)
- com.google.mlkit:face-detection - 16.0.0 -> 16.0.1
  - No public API changes
- com.google.mlkit:image-labeling - 16.0.0 -> 16.2.0
  - No public API changes
- com.google.mlkit:image-labeling-automl - 16.0.0 -> 16.2.0
  - No public API changes
- com.google.mlkit:image-labeling-common - 16.0.0 -> 16.2.0
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950913/Xamarin.Google.MLKit.ImageLabeling.Common.dll.diff.md.txt)
- com.google.mlkit:image-labeling-custom - 16.0.0 -> 16.2.0
  - No public API changes
- com.google.mlkit:language-id - 16.0.0 -> 16.1.0
  - No public API changes
- com.google.mlkit:object-detection - 16.0.0 -> 16.1.0
  - No public API changes
- com.google.mlkit:object-detection-common - 16.0.0 -> 16.1.0
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950928/Xamarin.Google.MLKit.ObjectDetection.Common.dll.diff.md.txt)
- com.google.mlkit:object-detection-custom - 16.0.0 -> 16.1.0
  - No public API changes
- com.google.mlkit:smart-reply - 16.0.0 -> 16.1.0
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950933/Xamarin.Google.MLKit.SmartReply.dll.diff.md.txt)
- com.google.mlkit:translate - 16.0.0 -> 16.1.0
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950937/Xamarin.Google.MLKit.Translate.dll.diff.md.txt)
- com.google.mlkit:vision-common - 16.0.0 -> 16.0.1
  - No public API changes
- com.google.mlkit:vision-internal-vkp - 16.0.0 -> 16.1.0
  - [breaking.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950945/Xamarin.Google.MLKit.Vision.Internal.Vkp.dll.breaking.md.txt)
  - [diff.md.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/5950946/Xamarin.Google.MLKit.Vision.Internal.Vkp.dll.diff.md.txt)

### Does this change any of the generated binding API's?

Yes. New API + changed API

### Describe your contribution

- new MLKit artifacts
- updates for older artifacts

Checklist
- [x] Add any metadata needed to get PR to compile
- [x] Verify any new namespaces are properly renamed
- [x] Locally generate diffs
- [x] Review any breaking changes
- [x] Attach diffs to PR

### CODE REVIEW

File: source/com.google.android.gms/play-services-mlkit-barcode-scanning/Transforms/Metadata.Namespaces.xml [hunk 1]
```diff
@@ -7,4 +7,27 @@
...
+        path="/api/package[@name='com.google.android.gms.internal.mlkit_vision_barcode']" 
+        name="managedName"
+        >
+        Android.Gms.Internal.MLKitVisionBarCode
+    </attr>
+    <attr 
+        path="/api/package[@name='com.google.mlkit.vision.barcode']" 
+        name="managedName"
+        >
+        Android.Gms.MLKit.Vision.BarCode
```
Comment (jpobst) [inline source/com.google.android.gms/play-services-mlkit-barcode-scanning/Transforms/Metadata.Namespaces.xml:20]: In https://github.com/xamarin/GooglePlayServicesComponents/pull/421, the namepace for `com.google.mlkit...` was changed to `Xamarin.Google.MLKit...`.  Here it is `Android.Gms.MLKit...`.

Is there a reason for the inconsistency?

Comment (moljac) [reply source/com.google.android.gms/play-services-mlkit-barcode-scanning/Transforms/Metadata.Namespaces.xml:20]: these packages from 2 different groupIds.

I know- confusing as hell!

**Comment by jpobst** [comment]
One question on namespaces, and CI fails.  (Not sure if CI needs updating?)

Other than that, looks good!

File: source/com.google.android.gms/play-services-mlkit-barcode-scanning/Transforms/Metadata.Namespaces.xml (hunk 1 above)
Comment (moljac) [reply source/com.google.android.gms/play-services-mlkit-barcode-scanning/Transforms/Metadata.Namespaces.xml:20]: Seem like they plan to move it from GMS to MLKit. 
I think you are right.
I hope people will not use both of those packages... Type collisions are expected to happen

File: source/com.google.mlkit/digital-ink-recognition/Transforms/Metadata.Namespaces.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        path="/api/package[@name='com.google.android.gms.internal.mlkit_vision_digitalink']" 
+        name="managedName"
+        >
+        Android.Gms.Internal.MLKitVisionDigitalInk
+    </attr>
+    <attr 
+        path="/api/package[@name='com.google.mlkit.vision.digitalink']" 
+        name="managedName"
+        >
+        Android.Gms.MLKit.Vision.DigitalInk
```
Comment (jpobst) [inline source/com.google.mlkit/digital-ink-recognition/Transforms/Metadata.Namespaces.xml:20]: Looks like 2 more `Xamarin.Google.MLKit` needed in this file.

**Review by jpobst — APPROVED** [review]

**Pull request merged**